controllers/bootstrap/init.go

```diff
@@ -485,49 +490,62 @@ func (b *Bootstrap) installODLM(operatorNs string, annotations map[string]string
 	// Install ODLM Operator
 	klog.Info("Installing ODLM Operator")
 	if operatorNs == constant.ClusterOperatorNamespace {
-		if err := b.createOrUpdateResource(annotations, OdlmClusterSubResource, b.MasterNamespace); err != nil {
+		if err := b.renderTemplate(constant.ODLMClusterSubscription, b.CSData); err != nil {
 			return err
 		}
 	} else {
-		if odlmSub := b.getYamlFromAnnotations(annotations, OdlmNamespacedSubResource); len(odlmSub) > 0 {
-			odlmSub = util.Namespacelize(odlmSub, placeholder, b.MasterNamespace)
-			odlmScopeEnable := "false"
-			// SaaS or on-prem multi instances case, enable odlm-scope
-			if b.MasterNamespace != b.ControlNs {
-				odlmScopeEnable = "true"
-			}
-			cm, err := util.GetCmOfMapCs(b.Client)
-			if err == nil {
-				err := util.UpdateNSList(b.Reader, b.Client, cm, "nss-odlm-scope", b.MasterNamespace, true)
-				if err != nil {
-					return err
-				}
-			} else if !errors.IsNotFound(err) {
-				klog.Errorf("Failed to get common-service-maps: %v", err)
-				return err
-			}
-			odlmSub = util.Namespacelize(odlmSub, placeholderOdlmScope, odlmScopeEnable)
-			if err := b.CreateOrUpdateFromYaml([]byte(odlmSub)); err != nil {
+		odlmScopeEnable := "false"
+		// SaaS or on-prem multi instances case, enable odlm-scope
+		if b.CSData.MasterNs != b.CSData.ControlNs {
+			odlmScopeEnable = "true"
+		}
+		cm, err := util.GetCmOfMapCs(b.Client)
+		if err == nil {
+			err := util.UpdateNSList(b.Reader, b.Client, cm, "nss-odlm-scope", b.CSData.MasterNs, true)
+			if err != nil {
 				return err
 			}
+		} else if !errors.IsNotFound(err) {
+			klog.Errorf("Failed to get common-service-maps: %v", err)
+			return err
+		}
+
+		odlmData := struct {
+			Channel           string
+			Version           string
+			MasterNs          string
+			ODLMScopeEnable   string
+			CatalogSourceName string
+			CatalogSourceNs   string
+		}{
+			Channel:           b.CSData.Channel,
+			Version:           b.CSData.Version,
+			MasterNs:          b.CSData.MasterNs,
+			ODLMScopeEnable:   odlmScopeEnable,
+			CatalogSourceName: b.CSData.CatalogSourceName,
+			CatalogSourceNs:   b.CSData.CatalogSourceNs,
+		}
+
+		if err := b.renderTemplate(constant.ODLMNamespacedSubscription, odlmData); err != nil {
+			return err
```

Comment:
I am wondering that whether we could use CSDATA instead of a new `odlmData`.

if we could integrate `ODLMScopeEnable` into `CSDATA` with predefined value `false`, and update the value at line 500 here
https://github.com/IBM/ibm-common-service-operator/pull/480/files#diff-d871a1a5fb2d0fae940235f9a8aff9610fa4009086d909ddcc7392047894f32bR500

Reply:
updated

Reply:
No, we don't need a default value.

In the ODLM, if env variable `ODLM_SCOPE` is not found, it will return `false` directly.